### PR TITLE
Feature/popover

### PR DESCRIPTION
**Pull request by NetanelBasal**

### CODE REVIEW

File: .idea/codeStyles/codeStyleConfig.xml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <state>
     <option name="USE_PER_PROJECT_SETTINGS" value="true" />
-    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Default" />
```
[the remark this answers is not in the document]
Comment (shaharkazaz) [reply .idea/codeStyles/codeStyleConfig.xml]: @bh86 isn't that from the theme?

File: playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      .pipe(take(1))
+      .subscribe(() => {
+        console.log('closed first');
+      });
+  }
+
+  open2(tpl: TemplateRef<any>) {
+    this.popover
+      .open(tpl, this.origin2.nativeElement, { className: 'my-popover' })
+      .pipe(take(1))
```
Comment (bh86) [inline playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.ts]: Why we need to put the `take` operator? Isn't the subscription always occur once?

File: playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      .subscribe(() => {
+        console.log('closed second');
```
Comment (bh86) [inline playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.ts]: I would consider to add a support to pass a result back to the subscription. Same as we did in the dialog (with the `DialogRef`). This way you'll be able to inject to your component the `DatoPanelRef` and close this specific popover instance and pass an optional result object.
Also a `data` object to pass to the embedded component (again, like the dialog).

File: playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.html [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+  <ng-template #tpl3>
+    <div [ngSwitch]="currentView">
+      <ng-container *ngSwitchCase="1">1</ng-container>
+      <ng-container *ngSwitchCase="2">2</ng-container>
+      <ng-container *ngSwitchCase="3">3</ng-container>
+      <ng-container *ngSwitchDefault>No view</ng-container>
+      <button (click)="switch(null)">Set default view</button>
+    </div>
+    <dato-button datoType="orange" datoPopoverClose>Close</dato-button>
```
Comment (bh86) [inline playground/src/app/components-gallery/previews/popover-preview/popover-preview.component.html]: Following my comment below, add an option to pass the result straight to the directive.

**Comment by NetanelBasal** [comment]
Merged into 2.2.0.

**Pull request closed without merging**